wsi/drm/drmbuffer.cpp

```diff
@@ -282,8 +306,17 @@ bool DrmBuffer::CreateFrameBuffer(uint32_t gpu_fd) {
   image_.drm_fd_ = 0;
   media_image_.drm_fd_ = 0;
 
-  int ret = drmModeAddFB2(gpu_fd, width_, height_, frame_buffer_format_,
-                          gem_handles_, pitches_, offsets_, &image_.drm_fd_, 0);
+  int ret = 0;
+  if(modifier_) {
```

Comment:
in drmplane, we would need to do something like this:
during initialization query what kind of modifier is supported for the format:

	for (int i = 0; i < (int)drm.plane->plane->count_formats; i++) {
		if (drm.plane->plane->formats[i] == DRM_FORMAT_XRGB8888)
			format_index = i;
	}

	for (int i = 0; i < (int)drm.plane->props->count_props; i++) {
		if (!strcmp(drm.plane->props_info[i]->name, "IN_FORMATS")) {
			drmModePropertyBlobPtr blob =
					drmModeGetPropertyBlob(drm.fd, drm.plane->props->prop_values[i]);

			struct drm_format_modifier_blob *m = blob->data;
			struct drm_format_modifier *mod = modifiers_ptr(m);

			for (int j = 0; j < (int)m->count_modifiers; j++, mod++) {
				if (mod->formats & (1ULL << format_index))
					drm.count_modifiers++;
			}
			drm.modifiers = calloc(drm.count_modifiers,
					       sizeof(uint64_t));
			mod = modifiers_ptr(m);
			for (int j = 0; j < (int)m->count_modifiers; j++, mod++) {
				if (mod->formats & (1ULL << format_index))
					drm.modifiers[j] = mod->modifier;
			}
		}
	}


You need to store that for each plane. You can check how we store it for rotation here:
https://github.com/intel/IA-Hardware-Composer/blob/master/wsi/drm/drmplane.cpp#L47

We can query from plane if modifier is supported before we try to create the offscreen buffer for that plane in displayplanemanager and also here.

Reply:
will do.

Reply:
You want to send this first as a separate PR.

Reply:
@kalyankondapally Do we really need to store the supported modifier for every supported format in this place?  drmModeAddFB2WithModifiers in the underlying driver will check if the modifiers are supported and return error if it is not. And if it is for RBC purpose, we already know DRM_FORMAT_XRGB8888/I915_FORMAT_MOD_Y_TILED_CCS is supported.

Reply:
Let's make the solution generic, so it's easily extendable for all platforms. i.e. platforms which dont have this support. So, ya please store it and than add API to query supported modifier for a given format.